core/controller/src/main/scala/whisk/core/entitlement/Entitlement.scala

```diff
@@ -208,8 +222,12 @@ protected[core] abstract class EntitlementProvider(
    * @param resource the resource the subject requests access to
    * @return a promise that completes with success iff the subject is permitted to access the requested resource
    */
-  protected[core] def check(user: Identity, right: Privilege, resource: Resource)(
-    implicit transid: TransactionId): Future[Unit] = check(user, right, Set(resource))
+  protected[core] def checkResource(
```

Comment:
I ran into an issue with these two `check()` methods, in that from my research it appears only one method is allowed to contain a default parameter at a given position in the method's signature [0]

I was unable to find a way around it short of more explicitly naming them. I'm open to suggestions for handling this scenario.

[0] https://groups.google.com/forum/#!msg/scala-user/FyQK3-cqfaY/fXLHr8QsW_0J